eng/common/scripts/modules/Package-Properties.psm1

```diff
@@ -0,0 +1,217 @@
+# Helper functions for retireving useful information from azure-sdk-for-* repo
+# Example Use : Import-Module .\eng\common\scripts\modules
+class PackageProps
+{
+    [string]$pkgName
+    [string]$pkgVersion
+    [string]$pkgDirectoryPath
+    [string]$pkgReadMePath
+    [string]$pkgChangeLogPath
+
+    PackageProps(
+        [string]$pkgName,
+        [string]$pkgVersion,
+        [string]$pkgDirectoryPath
+    )
+    {
+        $this.pkgName = $pkgName
+        $this.pkgVersion = $pkgVersion
+        $this.pkgDirectoryPath = $pkgDirectoryPath
+
+        if (Test-Path (Join-Path $pkgDirectoryPath "README.md")) {
+            $this.pkgReadMePath = Join-Path $pkgDirectoryPath "README.md"
+        } 
+        else {
+            $this.pkgReadMePath = $null
+        }
+
+        if (Test-Path (Join-Path $pkgDirectoryPath "CHANGELOG.md")) {
+            $this.pkgChangeLogPath = Join-Path $pkgDirectoryPath "CHANGELOG.md"
+        } 
+        else {
+            $this.pkgChangeLogPath = $null
+        }
+    }
+}
+
+Install-Module -Name powershell-yaml -RequiredVersion 0.4.1 -Force -Scope CurrentUser
+
+function Extract-PkgProps ($pkgPath, $pkgName)
+{
+    if ($Language -eq "net") {
+        Extract-DotNetPkgProps -pkgPath $pkgPath -pkgName $pkgName 
+    }
+    if ($Language -eq "java") {
+        Extract-JavaPkgProps -pkgPath $pkgPath -pkgName $pkgName 
+    }
+    if ($Language -eq "js") {
+        Extract-JsPkgProps -pkgPath $pkgPath -pkgName $pkgName 
+    }
+    if ($Language -eq "python") {
+        Extract-PythonPkgProps -pkgPath $pkgPath -pkgName $pkgName 
+    }
+}
+
+function Extract-DotNetPkgProps ($pkgPath, $pkgName)
+{
+    $projectPath = Join-Path $pkgPath "src" "$pkgName.csproj"
+    if (Test-Path $projectPath)
+    {
+        $projectData = New-Object -TypeName XML
+        $projectData.load($projectPath)
+        $pkgVersion = Select-XML -Xml $projectData -XPath '/Project/PropertyGroup/Version'
+        return [PackageProps]::new($pkgName, $pkgVersion, $pkgPath)
+    } else 
+    {
+        return $null
+    }
+}
+
+function Extract-JsPkgProps ($pkgPath, $pkgName)
+{
+    $projectPath = Join-Path $pkgPath "package.json"
+    if (Test-Path $projectPath)
+    {
+        $projectJson = Get-Content $projectPath | Out-String | ConvertFrom-Json
+        $jsStylePkgName = $pkgName.replace("azure-", "@azure/")
+        if ($projectJson.name -eq "$jsStylePkgName")
+        {
+            return [PackageProps]::new($projectJson.name, $projectJson.version, $pkgPath)
+        }
+    }
+    return $null
+}
+
+function Extract-PythonPkgProps ($pkgPath, $pkgName)
+{
+    $pkgName = $pkgName.Replace('_', '-')
+
+    if (Test-Path (Join-Path $pkgPath "setup.py"))
+    {
+        $setupLocation = $pkgPath.Replace('\','/')
+        pushd $RepoRoot
+        $setupProps = (python -c "import scripts.devops_tasks.common_tasks; obj=scripts.devops_tasks.common_tasks.parse_setup('$setupLocation'); print('{0},{1}'.format(obj[0], obj[1]));") -split ","
```

Comment:
We should probably make a note somewhere that python is required for parsing python packages.